In [2]:
import pickle
import numpy as np
import pandas as pd

In [3]:
model = pickle.load(open('concrete_strength.pkl','rb'))

In [4]:
df = pd.read_excel('Concrete_Data.xls')
df.head()

Cement (component 1)(kg in a m^3 mixture)  \
0                                      540.0   
1                                      540.0   
2                                      332.5   
3                                      332.5   
4                                      198.6   

   Blast Furnace Slag (component 2)(kg in a m^3 mixture)  \
0                                                0.0       
1                                                0.0       
2                                              142.5       
3                                              142.5       
4                                              132.4       

   Fly Ash (component 3)(kg in a m^3 mixture)  \
0                                         0.0   
1                                         0.0   
2                                         0.0   
3                                         0.0   
4                                         0.0   

   Water  (component 4)(kg in a m^3 mixture)  \
0                                      162.0   
1                                      162.0   
2                                      228.0   
3                                      228.0   
4                                      192.0   

   Superplasticizer (component 5)(kg in a m^3 mixture)  \
0                                                2.5     
1                                                2.5     
2                                                0.0     
3                                                0.0     
4                                                0.0     

   Coarse Aggregate  (component 6)(kg in a m^3 mixture)  \
0                                             1040.0      
1                                             1055.0      
2                                              932.0      
3                                              932.0      
4                                              978.4      

   Fine Aggregate (component 7)(kg in a m^3 mixture)  Age (day)  \
0                                              676.0         28   
1                                              676.0         28   
2                                              594.0        270   
3                                              594.0        365   
4                                              825.5        360   

   Concrete compressive strength(MPa, megapascals)   
0                                         79.986111  
1                                         61.887366  
2                                         40.269535  
3                                         41.052780  
4                                         44.296075

In [7]:
X = df.iloc[:,0:8].values
y = df.iloc[:,8:9].values

In [8]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [9]:
X_test

array([[ 266. ,  114. ,    0. , ...,  932. ,  670. ,  365. ],
       [ 362.6,  189. ,    0. , ...,  944.7,  755.8,    7. ],
       [ 389.9,  189. ,    0. , ...,  944.7,  755.8,   28. ],
       ...,
       [ 286.3,  200.9,    0. , ..., 1004.6,  803.7,    7. ],
       [ 424. ,   22. ,  132. , ...,  822. ,  750. ,   56. ],
       [ 339. ,    0. ,    0. , ...,  968. ,  781. ,   90. ]])

In [12]:
model.predict([X_test[0]])

array([51.69771027])

# Verification

In [13]:
sample = pd.DataFrame(X_test)

In [14]:
sample.head()

0      1      2      3     4      5      6      7
0  266.0  114.0    0.0  228.0   0.0  932.0  670.0  365.0
1  362.6  189.0    0.0  164.9  11.6  944.7  755.8    7.0
2  389.9  189.0    0.0  145.9  22.0  944.7  755.8   28.0
3  362.6  189.0    0.0  164.9  11.6  944.7  755.8    3.0
4  145.0    0.0  179.0  202.0   8.0  824.0  869.0   28.0

In [21]:
df[df['Cement (component 1)(kg in a m^3 mixture)'] == 266.0]

Cement (component 1)(kg in a m^3 mixture)  \
5                                        266.0   
8                                        266.0   
31                                       266.0   
61                                       266.0   
64                                       266.0   
906                                      266.0   

     Blast Furnace Slag (component 2)(kg in a m^3 mixture)  \
5                                                114.0       
8                                                114.0       
31                                               114.0       
61                                               114.0       
64                                               114.0       
906                                              112.0       

     Fly Ash (component 3)(kg in a m^3 mixture)  \
5                                           0.0   
8                                           0.0   
31                                          0.0   
61                                          0.0   
64                                          0.0   
906                                        87.0   

     Water  (component 4)(kg in a m^3 mixture)  \
5                                        228.0   
8                                        228.0   
31                                       228.0   
61                                       228.0   
64                                       228.0   
906                                      178.0   

     Superplasticizer (component 5)(kg in a m^3 mixture)  \
5                                                  0.0     
8                                                  0.0     
31                                                 0.0     
61                                                 0.0     
64                                                 0.0     
906                                               10.0     

     Coarse Aggregate  (component 6)(kg in a m^3 mixture)  \
5                                                932.0      
8                                                932.0      
31                                               932.0      
61                                               932.0      
64                                               932.0      
906                                              910.0      

     Fine Aggregate (component 7)(kg in a m^3 mixture)  Age (day)  \
5                                                670.0         90   
8                                                670.0         28   
31                                               670.0        365   
61                                               670.0        270   
64                                               670.0        180   
906                                              745.0         28   

     Concrete compressive strength(MPa, megapascals)   
5                                           47.029847  
8                                           45.854291  
31                                          52.908320  
61                                          51.732763  
64                                          48.695621  
906                                         39.417343

# index no: 31 has output of 52.908320. Our model predicted well.

# Our model gave a result of 51.69771027